clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -182,6 +182,10 @@ public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offs
 
     @Override
     public void commitTransaction() throws ProducerFencedException {
+        if (producerFencedOnCommitTxn) {
```

Comment:
We do not need to fence on close since we do not close producer upon suspending anymore: with onPartitionsLost KAFKA-7285 would not happen.